app/src/main/java/com/google/samples/apps/sunflower/PlantDetailFragment.kt

```diff
@@ -42,22 +46,42 @@ class PlantDetailFragment : Fragment() {
         val plantId = requireNotNull(arguments).getString(ARG_ITEM_ID)
 
         val factory = InjectorUtils.providePlantDetailViewModelFactory(requireActivity(), plantId)
-        val plantDetailViewModel = ViewModelProviders.of(this, factory)
-                .get(PlantDetailViewModel::class.java)
+        plantDetailViewModel = ViewModelProviders.of(this, factory)
+            .get(PlantDetailViewModel::class.java)
 
         val binding = DataBindingUtil.inflate<FragmentPlantDetailBinding>(
                 inflater, R.layout.fragment_plant_detail, container, false).apply {
             viewModel = plantDetailViewModel
             setLifecycleOwner(this@PlantDetailFragment)
             fab.setOnClickListener { view ->
-                plantDetailViewModel.addPlantToGarden()
-                Snackbar.make(view, R.string.added_plant_to_garden, Snackbar.LENGTH_LONG).show()
+                val isPlanted = plantDetailViewModel.isPlanted.value ?: false
+                if (isPlanted) {
+                    plantDetailViewModel.removePlantFromGarden()
+                    Snackbar.make(view, R.string.removed_plant_from_garden, Snackbar.LENGTH_LONG).show()
+                } else {
+                    plantDetailViewModel.addPlantToGarden()
+                    Snackbar.make(view, R.string.added_plant_to_garden, Snackbar.LENGTH_LONG).show()
+                }
+
             }
         }
 
+        subscribeUi(binding.root)
+
         return binding.root
     }
 
+    private fun subscribeUi(view: View) {
+        plantDetailViewModel.isPlanted.observe(viewLifecycleOwner, Observer {
+            val isPlanted = it ?: false
+            if (isPlanted) {
+                view.findViewById<FloatingActionButton>(R.id.fab).setImageResource(R.drawable.ic_minus)
```

Comment:
i believe you will cause a crash on pre-Lollipop devices if you directly set a Vector drawable resources like that

Reply:
I have run it in KitKat (Simulator) and it seems all right. I read the article "https://developer.android.com/guide/topics/graphics/vector-drawable-resources" and it says that "To change drawables at runtime, you can use the setImageResource() method as before.".
Could you give me more info about it?

Reply:
i've check the FAB source code, internally it intercepts and properly handle Vector drawables, so it wont crash. so you are correct.